### PR TITLE
Reviewed code

### PR DESCRIPTION
**Pull request by VladSelikh**

### CODE REVIEW

File: JS/Airport.js [hunk 1]
```diff
@@ -1,120 +1,83 @@
...
-     getPasPl() {
-        let pl = this.planes;
-        var x = [];
-        for (let p of this.planes) {
-            if (p instanceof PassengerPlane) {x.push(p);}
+    
+     getPassengerPlanes() {
+        var passengerPlaneList = [];
+        for (let plane of this.planes) {
+            if (plane instanceof PassengerPlane) {passengerPlaneList.push(plane);}
```
Comment (VladSelikh) [inline JS/Airport.js:12]: Try to rewrite this code using `filter()` array method.

File: JS/Airport.js [hunk 2]
```diff
@@ -1,120 +1,83 @@
...
-            if (plane instanceof MilitaryPlane) {//if
-                militaryPlanes.push(plane);
+            if (plane instanceof MilitaryPlane) {
+                militaryPlanesList.push(plane);
             }
-            //else
-            else{}
         });
-//return
-        return militaryPlanes;
```
Comment (VladSelikh) [inline JS/Airport.js:27]: Try to rewrite this code using `filter()` array method.

File: JS/Airport.js [hunk 3]
```diff
@@ -1,120 +1,83 @@
...
-        let militaryPlanes = this.getMilitaryPlanes();
-        for (let i = 0; i < militaryPlanes.length; i++) {
-        if (militaryPlanes[i].getMilitaryType() == MilitaryType.TYPE_TRANSPORT) {
-        transportMilitaryPlanes.push(militaryPlanes[i]);
-        }
-        }
-        return transportMilitaryPlanes;
+    getTransportMilitaryPlanes() {
+       let militaryPlanesArray = this.getMilitaryPlanes();
+       let transportMilitaryPlanes = militaryPlanesArray.filter(el => el.getMilitaryType() == MilitaryType.TRANSPORT);
```
Comment (VladSelikh) [inline JS/Airport.js:37]: You don't need `transportMilitaryPlanes` variable here.

File: JS/Airport.js [hunk 4]
```diff
@@ -1,120 +1,83 @@
...
     getBomberMilitaryPlanes()
     {
-        let bomberMilitaryPlanes = [];
         let militaryPlanes = this.getMilitaryPlanes();
-        for (let i = 0; i < militaryPlanes.length; i++) {
-            if (militaryPlanes[i].getMilitaryType()=== MilitaryType.BOMBER) {
-                bomberMilitaryPlanes.push(militaryPlanes[i]);
-            }
-        }
+        let bomberMilitaryPlanes = militaryPlanes.filter (el => el.getMilitaryType() == MilitaryType.BOMBER);
```
Comment (VladSelikh) [inline JS/Airport.js:44]: You don't need `bomberMilitaryPlanes` variable here.

File: JS/Airport.js [hunk 5]
```diff
@@ -1,120 +1,83 @@
...
         this.planes = planes;
     }
 
-
     getExperimentalPlanes() {
-        let experimentalPlanes  = [];
-        this.planes.forEach(plane => {
-            if (plane instanceof experimentalPlane) {//if
+        let experimentalPlanes = [];
+        this.planes.forEach((plane) => {
```
Comment (VladSelikh) [inline JS/Airport.js:54]: Try to rewrite this code using `filter()` array method.

File: JS/Planes/experimentalPlane.js [hunk 6]
```diff
@@ -1,68 +1,64 @@
-const Plane = require('./Plane');
```
Comment (VladSelikh) [inline JS/Planes/experimentalPlane.js:1]: Please rename file to have it started with capital letter.

File: JS/index.js [hunk 7]
```diff
@@ -1,37 +1,43 @@
...
-    console.log(`Passenger airport sorted by max speed: ${Airport.print(passengerAirport.sortByMaxSpeed())}`);
-    console.log(`Plane with max passenger capacity: ${Airport.print(passengerAirport.getPassengerPlaneWithMaxPassengersCapacity())}`);
-})();
\ No newline at end of file
+    let passengerAirport = new Airport(airport.getPassengerPlanes());
+    console.log(`Military airport sorted by max distance: ${print(militaryAirport.sortByMaxDistance())}`);
+    console.log(`Passenger airport sorted by max speed: ${print(passengerAirport.sortByMaxSpeed())}`);
+    console.log(`Plane with max passenger capacity: ${print(passengerAirport.getPassengerPlaneWithMaxPassengersCapacity())}`);
+    console.log(`list of military planes with transport type: ${print(militaryAirport.getTransportMilitaryPlanes())}`);
+    console.log(`List of Bomber planes In Military Planes: ${print(militaryAirport.getBomberMilitaryPlanes())}`);
```
Comment (VladSelikh) [inline JS/index.js:39]: Why did you add these 2 lines? There were no any in the original version.

File: JS/models/ExperimentalTypes.js [hunk 8]
```diff
@@ -1,8 +1,8 @@
 const ExperimentalTypes   = {
-    LIFTING_BODY: 'lifting_body',
+    LIFTING_BODY: 'LIFTING_BODY',
     HYPERSONIC: 'HYPERSONIC',
     HIGH_ALTITUDE: 'HIGH_ALTITUDE',
-    VTOL: "VTOL"
+    VTOL: "VTOL" // A vertical take-off and landing (VTOL) aircraft
```
Comment (VladSelikh) [inline JS/models/ExperimentalTypes.js:5]: Please do not use descriptive comments. The variable name should be descriptive by itself.

File: JS/models/ExperimentalTypes.js [hunk 9]
```diff
@@ -1,8 +1,8 @@
 const ExperimentalTypes   = {
```
Comment (VladSelikh) [inline JS/models/ExperimentalTypes.js:1]: Please use CAPITAL_LETTER notation for constants naming.

File: JS/models/classificationLevel.js [hunk 10]
```diff
@@ -1,8 +1,8 @@
 const ClassificationLevel  = {
```
Comment (VladSelikh) [inline JS/models/classificationLevel.js:1]: Please use CAPITAL_LETTER notation for constants naming.

File: JS/models/militaryType.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 const MilitaryType = {
```
Comment (VladSelikh) [inline JS/models/militaryType.js:1]: Please use CAPITAL_LETTER notation for constants naming.

File: JS/package.json [hunk 12]
```diff
@@ -4,7 +4,7 @@
   "description": "",
   "main": "",
   "scripts": {
-    "test": "mocha",
+    "test": "mocha ./test/airport.spec",
```
Comment (VladSelikh) [inline JS/package.json:7]: There is no need to change this line.

File: JS/utils.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+
+function print(planes) {
```
Comment (VladSelikh) [inline JS/utils.js:2]: Please give more meaningful name to the function.

File: JS/test/airport.spec.js [hunk 14]
```diff
@@ -1,104 +1,51 @@
...
         let nextPlaneMaxLoadCapacityIsHigherThanCurrent = true;
         for (let i = 0; i < planesSortedByMaxLoadCapacity.length - 1; i++) {
             let currentPlane = planesSortedByMaxLoadCapacity[i];
             let nextPlane = planesSortedByMaxLoadCapacity[i + 1];
-            if (currentPlane.getMinLoadCapacity() > nextPlane.getMinLoadCapacity()) {
+            if (currentPlane.getMaxLoadCapacity() > nextPlane.getMaxLoadCapacity()) {
                 nextPlaneMaxLoadCapacityIsHigherThanCurrent = false;
                 break;
             }
         }
```
Comment (VladSelikh) [inline JS/test/airport.spec.js:37]: Please rewrite this test without implementing any additional logic.

**Comment by VladSelikh** [comment]
You forgot to rename Planes folder in planes.

File: JS/Airport.js (hunk 4 above)
Comment (XenaF) [reply JS/Airport.js:44]: Done

File: JS/Airport.js (hunk 1 above)
Comment (XenaF) [reply JS/Airport.js:12]: Done

File: JS/Airport.js (hunk 2 above)
Comment (XenaF) [reply JS/Airport.js:27]: Done

File: JS/Airport.js (hunk 3 above)
Comment (XenaF) [reply JS/Airport.js:37]: Done

File: JS/Airport.js (hunk 5 above)
Comment (XenaF) [reply JS/Airport.js:54]: Done

File: JS/Planes/experimentalPlane.js (hunk 6 above)
Comment (XenaF) [reply JS/Planes/experimentalPlane.js:1]: Done

File: JS/index.js (hunk 7 above)
Comment (XenaF) [reply JS/index.js:39]: I practice how to add tests run it via 'npm run start'

File: JS/models/ExperimentalTypes.js (hunk 8 above)
Comment (XenaF) [reply JS/models/ExperimentalTypes.js:5]: Done. Variable was renamed throughout the project scripts

File: JS/models/ExperimentalTypes.js (hunk 9 above)
Comment (XenaF) [reply JS/models/ExperimentalTypes.js:1]: All constants in capital letters

File: JS/models/classificationLevel.js (hunk 10 above)
Comment (XenaF) [reply JS/models/classificationLevel.js:1]: All constants in capital letters

File: JS/models/militaryType.js (hunk 11 above)
Comment (XenaF) [reply JS/models/militaryType.js:1]: All constants in capital letters

File: JS/package.json (hunk 12 above)
Comment (XenaF) [reply JS/package.json:7]: line contains only 'mocha' word. Done

File: JS/utils.js (hunk 13 above)
Comment (XenaF) [reply JS/utils.js:2]: Done. 'Print' was renamed into 'printObjectToString'

File: JS/test/airport.spec.js (hunk 14 above)
Comment (XenaF) [reply JS/test/airport.spec.js:37]: Done

**Comment by VladSelikh** [comment]
Don't need this Pull Request anymore.
